### ISSUE TITLE
Execution context not being updated in nested context

### ISSUE THREAD

**Issue by SuyogSoti** [census-instrumentation/opencensus-python#716]
**Describe your environment.** Describe any aspect of your environment relevant to the problem, including your Python version, [platform]
I am running python 3.7 on windows.

**Steps to reproduce.**
Describe exactly how to reproduce the error. Include a code sample if applicable.
```python
from opencensus.trace.tracer import Tracer
from opencensus.trace import execution_context
from opencensus.trace.samplers import AlwaysOnSampler
import time

# Initialize a tracer, by default using the `PrintExporter`
tracer = Tracer(sampler=AlwaysOnSampler())

def do():
    time.sleep(0.05)

# Example for creating nested spans
with tracer.span(name='span1') as span1:
    with span1.span(name='span1_child1') as span1_child1:
        do()
    with span1.span(name='span1_child2') as span1_child2:
        print(execution_context.get_current_span().name)
        do()
with tracer.span(name='span2') as span2:
    do()
```

**What is the expected behavior?**
I expect it to print span1_child2

**What is the actual behavior?**
It prints span1

**Additional context.**
Not Really.

**Comment by reyang** [comment]
@SuyogSoti thanks for reporting this.

Would you try to change your code by using `tracer.span` instead of `span.span`?

```python
from opencensus.trace.tracer import Tracer
from opencensus.trace import execution_context
from opencensus.trace.samplers import AlwaysOnSampler
import time

# Initialize a tracer, by default using the `PrintExporter`
tracer = Tracer(sampler=AlwaysOnSampler())

# Example for creating nested spans
with tracer.span(name='span1') as span1:
    print(execution_context.get_current_span().name)
    with tracer.span(name='span1_child1') as span1_child1:
        print(execution_context.get_current_span().name)
    with tracer.span(name='span1_child2') as span1_child2:
        print(execution_context.get_current_span().name)
    print(execution_context.get_current_span().name)
with tracer.span(name='span2') as span2:
    print(execution_context.get_current_span().name)
```

@c24t it looks like we should update the doc and always use `tracer.span`, we don't know why we had `span.span`, and it seems to be broken. What do you think?

**Comment by reyang** [comment]
And probably we should remove `span.span` given it is broken? It is not clear to me that why we had it in the first place. Technically this is a breaking API change (by removing a broken API) though.

**Comment by SuyogSoti** [comment]
Thanks! I will use `tracer.span`

**Comment by reyang** [comment]
We'll fix the document in #733 to mitigate this issue.

The API improvement will happen in [OpenTelemetry Python SDK](https://github.com/open-telemetry/opentelemetry-python).
